enhancements/machine-api/machine-deletion-hooks.md

```diff
@@ -238,18 +282,23 @@ extenuating circumstances) may decide to remove a particular lifecycle hook
 to allow the machine controller to progress past the corresponding lifecycle-point.
 
 The Machine status will contain conditions identifying why the Machine controller
-has not progressed in removing the Machine. It is expected that hook implementing
-controllers should signal to users when they are having issues via some mechanism
-(eg. their `ClusterOperator` status) and that the conditions on the Machine should
-identify the components which the user should look at in order to determine any
-underlying issues.
-
-For example, the condition will contain the owner of the hook blocking the removal,
-the user should look towards the owner's own reporting to identify issues.
-Any OpenShift operator experiencing issues should already have a mechanism in which
-it can report errors and having these errors follow the normal mechanisms should
-place the failure reporting closer to the issue, as opposed to placing it directly
-onto the Machine.
+has not progressed in removing the Machine. This will be in the form of new `Drainable`
+and `Terminable` conditions.
+
+It is expected that hook implementing controllers should signal to users when they
+are having complete system issues via some mechanism (eg. their `ClusterOperator` status if the issue is endangering
+the cluster health) and that they should also use conditions on the Machine to
+identify issues with removing the hook from a particular Machine.
+
+When errors occur, hook implementing controllers are encouraged to add a condition
+to Machines with details of the error. The condition `Type` should be formad as the
+hook name, prefixed by the lifecycle point. For example, for a `preDrain` hook with name
+`MigrateImportantApp`, the condition `Type` would be `PreDrainMigrateImportantApp`.
```

Comment:
`PreDrainMigrateImportantAppSucceeded` ?  So there's a clear meaning to true an false.